Doc/c-api/sys.rst

```diff
@@ -222,10 +222,55 @@ These are utility functions that make functionality from the :mod:`sys` module
 accessible to C code.  They all work with the current interpreter thread's
 :mod:`sys` module's dict, which is contained in the internal thread state structure.
 
+.. c:function:: PyObject *PySys_GetAttr(PyObject *name)
+
+   Return the object *name* from the :mod:`sys` module or ``NULL`` on failure.
+   Set :exc:`RuntimeError` and return ``NULL`` if it does not exist.
+
+   If the non-existing object should not be treated as a failure, you can use
+   :c:func:`PySys_GetOptionalAttr` instead.
+
+   .. versionadded:: 3.13
+
+.. c:function:: PyObject *PySys_GetAttrString(const char *name)
+
+   This is the same as :c:func:`PySys_GetAttr`, but *name* is
+   specified as a :c:expr:`const char*` UTF-8 encoded bytes string,
+   rather than a :c:expr:`PyObject*`.
+
+   If the non-existing object should not be treated as a failure, you can use
+   :c:func:`PySys_GetOptionalAttrString` instead.
+
+   .. versionadded:: 3.13
+
+.. c:function:: int PySys_GetOptionalAttr(PyObject *name, PyObject **result);
+
+   Variant of :c:func:`PySys_GetAttr` which doesn't raise
+   exception if the object does not exist.
+
+   If the object exists, set *\*result* to a new :term:`strong reference`
+   to the object and return ``1``.
+   If the object does not exist, set *\*result* to ``NULL`` and return ``0``,
+   without setting an exception.
+   If other error occurred, set an exception, set *\*result* to ``NULL`` and
+   return ``-1``.
```

Comment:
I suggest to use a list and start with the return value, so it's easier to see the 3 cases:

```suggestion
   * Return ``1`` and set *\*result* to a new :term:`strong reference`
     to the object if the attribute exists.
   * Return ``0`` without setting an exception and set *\*result* to ``NULL``
     if the attribute does not exist.
   * Return ``-1``, set an exception and set *\*result* to ``NULL``
     if an error occurred.
```

Reply:
"Return and then set exception and variable"  looks like a wrong sequence to me. It cannot do anything after returning.